spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/elasticsearch/rest/RestClientAutoConfiguration.java

```diff
@@ -92,4 +93,17 @@ public RestHighLevelClient restHighLevelClient(RestClientBuilder restClientBuild
 
 	}
 
+	@Configuration(proxyBeanMethods = false)
+	@ConditionalOnClass(RestHighLevelClient.class)
+	@ConditionalOnBean(RestHighLevelClient.class)
+	public static class RestClientConfiguration {
+
+		@Bean
+		@ConditionalOnMissingBean
+		public RestClient restClient(RestHighLevelClient restHighLevelClient) {
```

Comment:
There's already another bean method named for `RestClient` with the same name. Wouldn't that create issues with bean overriding?
Maybe we could change the existing bean method and inject `ObjectProvider<RestHighLevelClient>` and do everything there (return the client from the high level if it exists or create a new one from the builder). What fo you think (I've quickly looked at your PR, so I might haved missed something)?